packages/loader/driver-utils/src/parallelRequests.ts

```diff
@@ -487,6 +494,10 @@ async function getSingleOpBatch(
 			}
 		}
 
+		const waitStartTime = performance.now();
+		telemetryEvent = PerformanceEvent.start(logger, {
```

Comment:
Given this is in while-loop, you should make this conditional, i.e. create an object only if we did not create it on previous iterations. That way we will capture all wait time through one event instance (which is I believe intention here).